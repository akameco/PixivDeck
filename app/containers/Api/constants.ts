import { API_REQUEST_SUCCESS_TYPE } from './actionTypes'

export const API_REQUEST_SUCCESS: API_REQUEST_SUCCESS_TYPE =
  'Api/API_REQUEST_SUCCESS'
