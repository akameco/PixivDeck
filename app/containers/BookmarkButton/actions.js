// @flow
import type { Action } from './actionTypes'
import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILER,
  DELETE_BOOKMARK_FAILER,
} from './constants'
import type { Restrict } from './types'

export function addBookmarkRequest(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}

export function addBookmarkSuccess(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}

export function deleteBookmarkRequest(id: number): Action {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}

export function deleteBookmarkSuccess(id: number): Action {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}

export function addBookmarkFailer(id: number, error: string): Action {
  return {
    type: ADD_BOOKMARK_FAILER,
    id,
    error,
  }
}

export function deleteBookmarkFailer(id: number, error: string): Action {
  return {
    type: DELETE_BOOKMARK_FAILER,
    id,
    error,
  }
}
