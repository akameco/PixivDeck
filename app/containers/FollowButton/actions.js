// @flow
import type { Action } from './actionTypes'
import {
  FOLLOW_REQUEST,
  UN_FOLLOW_REQUEST,
  FOLLOW_SUCCESS,
  UN_FOLLOW_SUCCESS,
  FOLLOW_FAILER,
  UN_FOLLOW_FAILER,
} from './constants'

export function followRequest(
  id: number,
  restrict: 'public' | 'private'
): Action {
  return {
    type: FOLLOW_REQUEST,
    id,
    restrict,
  }
}

export function unFollowRequest(
  id: number,
  restrict: 'public' | 'private'
): Action {
  return {
    type: UN_FOLLOW_REQUEST,
    id,
    restrict,
  }
}

export function followSuccess(restrict: 'public' | 'private'): Action {
  return {
    type: FOLLOW_SUCCESS,
    restrict,
  }
}

export function unFollowSuccess(restrict: 'public' | 'private'): Action {
  return {
    type: UN_FOLLOW_SUCCESS,
    restrict,
  }
}

export function followFailer(error: string): Action {
  return {
    type: FOLLOW_FAILER,
    error,
  }
}

export function unFollowFailer(error: string): Action {
  return {
    type: UN_FOLLOW_FAILER,
    error,
  }
}
