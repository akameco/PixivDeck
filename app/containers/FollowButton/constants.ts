import {
  FOLLOW_REQUEST_TYPE,
  FOLLOW_SUCCESS_TYPE,
  FOLLOW_FAILER_TYPE,
  UN_FOLLOW_REQUEST_TYPE,
  UN_FOLLOW_SUCCESS_TYPE,
  UN_FOLLOW_FAILER_TYPE,
} from './actionTypes'

export const FOLLOW_REQUEST: FOLLOW_REQUEST_TYPE = 'FollowButton/FOLLOW_REQUEST'
export const FOLLOW_SUCCESS: FOLLOW_SUCCESS_TYPE = 'FollowButton/FOLLOW_SUCCESS'
export const FOLLOW_FAILER: FOLLOW_FAILER_TYPE = 'FollowButton/FOLLOW_FAILER'
export const UN_FOLLOW_REQUEST: UN_FOLLOW_REQUEST_TYPE =
  'FollowButton/UN_FOLLOW_REQUEST'
export const UN_FOLLOW_SUCCESS: UN_FOLLOW_SUCCESS_TYPE =
  'FollowButton/UN_FOLLOW_SUCCESS'
export const UN_FOLLOW_FAILER: UN_FOLLOW_FAILER_TYPE =
  'FollowButton/UN_FOLLOW_FAILER'
