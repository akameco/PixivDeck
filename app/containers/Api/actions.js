// @flow
import type { Response } from 'services/api'
import type { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
