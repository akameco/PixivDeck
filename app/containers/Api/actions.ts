import { Response } from 'services/api'
import { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
