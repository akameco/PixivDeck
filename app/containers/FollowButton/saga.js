// @flow
import type { Saga } from 'redux-saga'
import { call, takeEvery, put } from 'redux-saga/effects'
import * as api from '../Api/sagas'
import * as actions from './actions'
import * as Actions from './constants'

type Props = {
  id: number,
}

export function* follow({ id }: Props): Saga<void> {
  try {
    const data = { userId: id, restrict: 'public' }
    yield call(api.post, '/v1/user/follow/add', data)
    yield put(actions.followSuccess('public'))
  } catch (error) {
    yield put(actions.followFailer(error))
  }
}

export function* unfollow({ id }: Props): Saga<void> {
  try {
    const data = { userId: id, restrict: 'public' }
    yield call(api.post, '/v1/user/follow/delete', data)
    yield put(actions.unFollowSuccess('public'))
  } catch (error) {
    yield put(actions.unFollowFailer(error))
  }
}

function* root(): Saga<void> {
  yield takeEvery(Actions.FOLLOW_REQUEST, follow)
  yield takeEvery(Actions.UN_FOLLOW_REQUEST, unfollow)
}

export default root
