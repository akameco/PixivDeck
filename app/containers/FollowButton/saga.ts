import { call, takeEvery, put } from 'redux-saga/effects'
import * as api from '../Api/sagas'
import * as actions from './actions'
import * as Actions from './constants'

interface Props {
  id: number
}
export function* follow({ id }: Props) {
  try {
    const data = {
      userId: id,
      restrict: 'public',
    }
    yield call(api.post, '/v1/user/follow/add', data)
    yield put(actions.followSuccess('public'))
  } catch (error) {
    yield put(actions.followFailer(error))
  }
}
export function* unfollow({ id }: Props) {
  try {
    const data = {
      userId: id,
      restrict: 'public',
    }
    yield call(api.post, '/v1/user/follow/delete', data)
    yield put(actions.unFollowSuccess('public'))
  } catch (error) {
    yield put(actions.unFollowFailer(error))
  }
}

function* root() {
  yield takeEvery(Actions.FOLLOW_REQUEST, follow)
  yield takeEvery(Actions.UN_FOLLOW_REQUEST, unfollow)
}

export default root
