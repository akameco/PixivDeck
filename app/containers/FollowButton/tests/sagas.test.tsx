import { takeEvery, put } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as constants from '../constants'
import * as actions from '../actions'

test('root', () => {
  const gen = sagas.default()
  let next = gen.next()
  expect(next.value).toStrictEqual(
    takeEvery(constants.FOLLOW_REQUEST, sagas.follow)
  )
  next = gen.next()
  expect(next.value).toStrictEqual(
    takeEvery(constants.UN_FOLLOW_REQUEST, sagas.unfollow)
  )
})

test('follow success', () => {
  const gen = sagas.follow({
    id: 1,
  })
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.next()
  expect(next.value).toStrictEqual(put(actions.followSuccess('public')))
})

test('follow failer', () => {
  const gen = sagas.follow({
    id: 1,
  })
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.throw('err')
  expect(next.value).toStrictEqual(put(actions.followFailer('err')))
})

test('unfollow success', () => {
  const gen = sagas.unfollow({
    id: 1,
  })
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.next()
  expect(next.value).toStrictEqual(put(actions.unFollowSuccess('public')))
})

test('unfollow failer', () => {
  const gen = sagas.unfollow({
    id: 1,
  })
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.throw('err')
  expect(next.value).toStrictEqual(put(actions.unFollowFailer('err')))
})
