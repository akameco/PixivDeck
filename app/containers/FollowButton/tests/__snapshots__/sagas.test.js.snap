// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`follow failer 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v1/user/follow/add",
      Object {
        "restrict": "public",
        "userId": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`follow success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v1/user/follow/add",
      Object {
        "restrict": "public",
        "userId": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`unfollow failer 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v1/user/follow/delete",
      Object {
        "restrict": "public",
        "userId": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`unfollow success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v1/user/follow/delete",
      Object {
        "restrict": "public",
        "userId": 1,
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
